addons/Trusts/Trusts.lua

```diff
@@ -0,0 +1,338 @@
+  end
+  return limit;
+end
+
+function call_trust()
+  if #queue > 0 then
+    windower.send_command('input /ma "'..windower.to_shift_jis(check_lang(queue[1]))..'" <me>')
+  end
+end
+
+function check_exist()
+  local party = {} --include only trusts. --['name']=models
+  local party_ind = {} -- index of trust's name in current party. {'name1','name2',...,'name5'}
+  for i=1,5 do
+    local member = windower.ffxi.get_party()['p'..i]
+    if member then
+      if member.mob.spawn_type == 14 then
+        party[member.name]= member.mob.models[1]
+        table.insert(party_ind,member.name)
+      end
+    end
+  end
+  return {party,party_ind};
+end
+
+function call_set(set)
+  queue = {} --trusts to be cast.
+  settings = config.load()
+  local party,party_ind = unpack(check_exist())
+  local limit = check_limit() --upper limit # of calls trust in current player.
+  local time = os.clock() --window open
+
+  if set == 'random' then
+    local checked = {}
+    local others = windower.ffxi.get_party().party1_count - #party_ind - 1 --# of human in party exept <me>.
+
+    if limit == #party_ind then
+      windower.send_command('input /retr all')
+      calls = limit
+      coroutine.sleep(settings.wait.retrall)
+    else
+      calls = limit - #party_ind - others
+    end
+
+    repeat
+      local index = trusts[math.random(1,#trusts)]
+      if not table.find(checked,index.name) then
+        table.insert(checked,index.name)
+        if windower.ffxi.get_spells()[index.id] and windower.ffxi.get_spell_recasts()[index.id] == 0 then
+          table.insert(queue,index)
+        end
+      end
+    until #queue >= calls or #checked >= 103 --# of unique names w/o Cornelia
+
+  elseif settings.sets[set] then
+    retr = {unpack(party_ind)}
+    for i=1,limit do
+      if settings.sets[set][tostring(i)] then
+        local entity = trusts:with(lang,settings.sets[set][tostring(i)])
+        if not party[entity.name]
+        or party[entity.name] ~= entity.models then
+          if windower.ffxi.get_spell_recasts()[entity.id] == 0 then
+            if windower.ffxi.get_spells()[entity.id] then
+              table.insert(queue,entity)
+            else
+              table.remove(retr,table.find(retr,party_ind[i]))
+              error('You aren\'t trusted by '..entity.english..'.')
+            end
+          else
+            table.remove(retr,table.find(retr,party_ind[i]))
+            local recast = math.floor(windower.ffxi.get_spell_recasts()[entity.id] / 6) / 10
+            log(entity.english..' needs '..recast..' secs break.')
+          end
+        else
+          table.remove(retr,table.find(retr,entity.name))
+          if settings.auto then
+            log(entity.english..' already exists.')
+          end
+        end
+      end
+    end
+    for index,name in pairs(retr) do
+      if #retr == #party_ind then
+        windower.send_command('input /retr all')
```

Comment:
For game-inputs you should always use `windower.chat.input('/retr all')`, it's not only faster but also "safer" in that fewer things can go wrong with conversions.